cli/azd/internal/cmd/add/add_configure.go

```diff
@@ -0,0 +1,209 @@
+package add
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+	"unicode"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/fatih/color"
+)
+
+type promptCtx struct {
+	context.Context
+	prj *project.ProjectConfig
+}
+
+// configure fills in the fields for a resource.
+func configure(
+	r *project.ResourceConfig,
+	console input.Console,
+	ctx promptCtx) (*project.ResourceConfig, error) {
+	switch r.Type {
+	case project.ResourceTypeHostContainerApp:
+		return fillUses(r, console, ctx)
+	case project.ResourceTypeOpenAiModel:
+		return fillAiModelName(r, console, ctx)
+	case project.ResourceTypeDbPostgres,
+		project.ResourceTypeDbMongo:
+		return fillDatabaseName(r, console, ctx)
+	case project.ResourceTypeDbRedis:
+		r.Name = "redis"
```

Comment:
Sort of surprised not to see a `fillRedisName`?  Do we not need to prompt the user for the name of this?  Is the assumption that the user would not want to control the name for some reason?

Reply:
I could see from an infrastructure management perspective and knowing how Azure works; it may be nice to have a name provided.

From an app developer perspective, I wouldn't expect to be prompted for a redis database name; generally, redis databases are unnamed and numbered: https://www.digitalocean.com/community/cheatsheets/how-to-manage-redis-databases-and-keys.

We don't support N-databases yet (it'd be important to work through what this actually looks like from an infra perspective), so the user doesn't lose any functionality with the current flow.

What do you think an experience for Redis naming would look like?

Reply:
> We don't support N-databases yet (it'd be important to work through what this actually looks like from an infra perspective), so the user doesn't lose any functionality with the current flow.
> 
> What do you think an experience for Redis naming would look like?

Not sure - the thing I sort of wondered about was at what level this name needs to be unique?  I was naively assuming that this would influence the physical name and that without the user being able to control it you'd only be able to create one of these per subscription?

If we aren't going to run into uniqueness issues on the name then I'm fine deferring for now - I agree that having multiple redis resources would be uncommon.

Reply:
Added validation for this, track the item with #4538